### PR TITLE
 Replace the youtube API with web scrapping

### LINKED ISSUE #21: Replace the youtube API with web scrapping
**Is your feature request related to a problem? Please describe.**
Currently, data API is used with the App and since the calls are not completely free it would be sensible to replace the API with scrapping as it is just used to get video ids

**Describe the solution you'd like**
Use some library like beautiful soup to scrape the vid's from youtube

**Additional context**
One of the highest priority issues.

### PR DESCRIPTION
**Pull request by SLakhani1**
Fixes #21 
Used web scrapping as replacement of API.

## Description
youtube_search(): It searches the query on youtube and returns the links of all the results.
youtube_related(): It opens the first link returned by youtube_search() and returns the links of all the videos present in the related videos section.

## Motivation and Context

Issue: #21 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: server/modules/youtube_videos.py [hunk 1]
```diff
@@ -1,75 +1,44 @@
-# API , video_id documentation : https://gist.github.com/dgp/1b24bf2961521bd75d6c
-
-from apiclient.discovery import build
-from apiclient.errors import HttpError
-
-
-DEVELOPER_KEY = "PASTE_YOUR_API_KEY_HERE"
-YOUTUBE_API_SERVICE_NAME = "youtube"
-YOUTUBE_API_VERSION = "v3"
+import webbrowser, requests, sys, bs4, re
```
Comment (sansyrox) [inline server/modules/youtube_videos.py]: Please shift the imports to a single line. It improves the readability of code.

File: server/modules/youtube_videos.py [hunk 2]
```diff
@@ -1,75 +1,44 @@
...
+        regex2 = re.compile(r'/(watch\?v=|embed/|v/|.+\?v=)?(?P<id>[A-Za-z0-9\-=_]{11})')
+        VideosId2 = regex2.match(videoLinks2[i].get('href'))
+        if not VideosId2:
+            print('no match')
+        else:
+            rel_videos.append([videoLinks2[i].get('title'),VideosId2.group('id')])
+    return rel_videos
+
+
+
```
Comment (sansyrox) [inline server/modules/youtube_videos.py]: Remove the extra lines, leaving an eof.

**Review by sansyrox — CHANGES_REQUESTED** [review]
Please check the inline comments. 
Otherwise, a great job with the PR 🚀 🚀

File: server/modules/youtube_videos.py (hunk 2 above)
Comment (SLakhani1) [reply server/modules/youtube_videos.py]: Done

**Review by sansyrox — APPROVED** [review]

**Pull request merged**